### DISCUSSION TITLE
Empty web page, excel file is not created

### DISCUSSION THREAD

**Discussion by tiresiaSpartano** [PHPOffice/PhpSpreadsheet#3449] [Q&A]
Hi everyone. I'm a beginner.

I have a ubuntu 22.04.2 machine with PHP 8.1.2. I installed Phpspreadsheet with Composer. Apparently without errors and satisfying all dependencies. 

I then create a new project in /var/www/html/test. In this directory I have composer.lock composer.json and vendor/ subdirectory.
Always inside test/ I have the file test.php, containing this script:

`<?php
use PhpOffice\PhpSpreadsheet\Spreadsheet;
use PhpOffice\PhpSpreadsheet\Writer\Xlsx;

$spreadsheet = new Spreadsheet();
$sheet = $spreadsheet->getActiveSheet();
$sheet->setCellValue('A1', 'Hello World!');

$writer = new Xlsx($spreadsheet);
$writer->save('hello_world.xlsx');
?>`

When i run the script, i get a blank web page and the excel file is not created.
What am I doing wrong?
Thanks.

**Comment by MarkBaker** [comment]
There is no screen output in this script, so it will show a blank page when you run it.
The line `$writer->save('hello_world.xlsx');` is telling PHP to save the spreadsheet as a file called 'hello_world.xlsx' on your web server. The file is being created, and you should find it in your `/var/www/html/test` folder.

**Comment by tiresiaSpartano** [comment]
Hi there. Thanks for the reply.
Unfortunately, the excel file is not generated in the directory, as I wrote in the title.
I checked for any unmet dependencies with the command

`composer check-platform-reqs`

This is the output:

Checking platform requirements for packages in the vendor dir
ext-ctype     8.1.2      success
ext-dom       20031129   success
ext-fileinfo  8.1.2      success
ext-gd        8.1.2      success
ext-iconv     8.1.2      success
ext-json      8.1.2      success
ext-libxml    8.1.2      success
ext-mbstring  8.1.2      success
ext-simplexml 8.1.2      success
ext-xml       8.1.2      success
ext-xmlreader 8.1.2      success
ext-xmlwriter 8.1.2      success
ext-zip       1.19.5     success
ext-zlib      8.1.2      success
php           8.1.2      success

I don't understand where I'm going wrong.

**Comment by MarkBaker** [comment]
If the code wasn't running correctly, then you would see a page with errors. That means the library has been installed correctly, otherwise you would see errors like `Spreadsheet class not found`; and the file is being created somewhere, otherwise you would see errors about being unable to create the file.
You don't specify a directory path in your call to save(), only the filename, so the file should be saved in PHP's current working directory

What I don't see in your code is the include/require for the composer autoload

**Comment by tiresiaSpartano** [comment]
Sorry Mark, You are right. Here is the complete script

`<?php

require 'vendor/autoload.php';

use PhpOffice\PhpSpreadsheet\Spreadsheet;
use PhpOffice\PhpSpreadsheet\Writer\Xlsx;

$spreadsheet = new Spreadsheet();
$sheet = $spreadsheet->getActiveSheet();
$sheet->setCellValue('A1', 'Hello World !');

$writer = new Xlsx($spreadsheet);
$writer->save('hello world.xlsx');
?>`

I would expect to find the xlsx file in this directory: /var/www/html/test. And instead there is nothing.

As you say, I'm not getting any error messages. So, apparently, everything is working out.

**Comment by MarkBaker** [comment]
You'd have got some errors running that script if it hadn't been there ;-)

Try adding some diagnostic echo outputs to show each step; include a dump of `getcwd()` to confirm the current working directory; and try using a full filepath, so that you can be certain what directory the file is to be saved in

**Comment by tiresiaSpartano** [comment]
I tried to introduce the `getcwd()` function.It works anywhere in the script, except after the `save()` function. In this case the path is not displayed.

`<?php

require 'vendor/autoload.php';

use PhpOffice\PhpSpreadsheet\Spreadsheet;
use PhpOffice\PhpSpreadsheet\Writer\Xlsx;

$spreadsheet = new Spreadsheet();
$sheet = $spreadsheet->getActiveSheet();
$sheet->setCellValue('A1', 'Hello World !');

$writer = new Xlsx($spreadsheet);

$dir = getcwd();
echo "Current directory: " . $dir;

$writer->save('/var/www/html/test/hello_world.xlsx');

?>`

As you can see I also tried using an absolute path to save the file, but that didn't work.

**Reply by MarkBaker** [reply]
In that case, I haven't got a clue! The file is being saved somewhere, otherwise you'd be seeing errors. It should be saved in your current directory if you don't specify a path as well as the filename; when you specify an absolute path, it should save the file in that folder.

Try sending it to the browser instead, so that it will download when you run the script (sending the appropriate headers). Make sure you remove any output from the script first though, including any new lines after that `?>` (best get rid of the `?>` completely).
